### DISCUSSION TITLE
Custom resolver not updating formState: {errors} on dynamic validation

### DISCUSSION THREAD

**Discussion by mburisin** [react-hook-form/react-hook-form#8734] [General]
### Version Number

7.33.1

### Codesandbox/Expo snack

https://codesandbox.io/s/custom-hook-with-resolver-v7-forked-3rvv0i

### Steps to reproduce

1. write some text in firstName input
2. delete text

### Expected behaviour

I'm trying to get custom yup resolver to work, but I have problems with getting correct values from formState: {errors}

Mode is set to 'all' so errors should be updated dynamically. Now formState: { error } state is updated according to resolver output only on submit.

I tried to use trigger function without any luck as well.

I based the example on this docs https://react-hook-form.com/advanced-usage#CustomHookwithResolver, 
https://codesandbox.io/s/custom-hook-with-resolver-v7-cwczk, but I had to change the code a bit, because it didn't seem to work properly - no errors were reported to console, there is errors include instead of formState: {errors}.

Am I doing something wrong?

This resolver without Yup integration seems to work properly https://codesandbox.io/s/react-hook-form-resolver-rvspp

Many thanks for any input!

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by bluebill1049** [comment]
could you try to follow this example: https://codesandbox.io/s/react-hook-form-customresoliver-v7-8mpqw ?

**Reply by LydiaF** [reply]
Hey @bluebill1049,

I also ran into this issue. Here is a code sandbox of Yup and `trigger()` always returning undefined, regardless of whether the form is valid or not. https://codesandbox.io/s/custom-hook-with-resolver-v7-forked-1v652p?file=/src/App.js

I tried your code sandbox with Joi using trigger, modified it to use `trigger()` and it worked as expected. https://codesandbox.io/s/react-hook-form-customresoliver-v7-forked-cyxo88?file=/src/index.jsx

Is the solution here/Do you recommend using Joi, and not Yup?

Thanks for any advice :)

**Reply by leapful** [reply]
@LydiaF 

Please do not ask question on an unrelated discussion like this. Your question is related to `trigger` method while this topic is about form state errors is not updated.

In the next time, please pay attention on the version of react-hook-form library too. You're using an old version of react-hook-form therefore trigger validation result is not expected. Just use the latest version that support the new validation handling to make it work.

Corrected sample: https://codesandbox.io/s/custom-hook-with-resolver-v7-forked-hu8uf0

**Comment by juanmespina** [comment]
I get the same issue. It's a custom resolver, I console.log at the end of the resolver and it cleans the data but formState keeps the same error
